trinity/sync/beacon/chain.py

```diff
@@ -66,55 +65,32 @@ def __init__(self,
 
         self.sync_peer = None
 
-    @property
-    def is_sync_peer_selected(self) -> bool:
-        return self.sync_peer is not None
-
     async def _run(self) -> None:
-        for retry in itertools.count():
-            is_last_retry = retry == PEER_SELECTION_MAX_RETRIES - 1
-            if retry >= PEER_SELECTION_MAX_RETRIES:
-                raise Exception("Invariant: Cannot exceed max retries")
-
+        while True:
             try:
                 self.sync_peer = await self.wait(self.select_sync_peer())
             except ValidationError as exception:
                 self.logger.info(f"No suitable peers to sync with: {exception}")
-                if is_last_retry:
-                    # selecting sync peer has failed
-                    break
-                else:
-                    # wait some time and try again
-                    await asyncio.sleep(PEER_SELECTION_RETRY_INTERVAL)
-                    continue
-            else:
-                # sync peer selected successfully
-                break
-
-            raise Exception("Unreachable")
-
-        if not self.is_sync_peer_selected:
-            self.logger.info("Failed to find suitable sync peer in time")
-            return
-
-        await self.wait(self.sync())
-
-        new_head = await self.chain_db.coro_get_canonical_head(BeaconBlock)
-        self.logger.info(f"Sync with {self.sync_peer} finished, new head: {new_head}")
+                # wait some time and try again
+                await asyncio.sleep(PEER_SELECTION_RETRY_INTERVAL)
+                continue
+
+            # sync peer selected successfully
+            await self.wait(self.sync())
+            new_head = await self.chain_db.coro_get_canonical_head(BeaconBlock)
+            self.logger.info(f"Sync with {self.sync_peer} finished, new head: {new_head}")
```

Comment:
Should not use f-string for logging.
```suggestion
            self.logger.info(
                "Sync with %s finished, new head: %s",
                self.sync_peer,
                new_head,
            )
```